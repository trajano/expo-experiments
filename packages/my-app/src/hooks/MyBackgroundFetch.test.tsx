import { act, render, renderHook } from '@testing-library/react-native';
import { Text } from 'react-native';
import {
  MyBackgroundFetchContext,
  MyBackgroundFetchProvider,
  useMyBackgroundFetch,
  WithMyBackgroundFetch,
} from './MyBackgroundFetch';
import { BackgroundFetchStatus } from 'expo-background-fetch';
import { PropsWithChildren } from 'react';

const TestComponent: React.FC = () => {
  const MyBackgroundFetchProps = useMyBackgroundFetch();

  return <Text testID="props">{JSON.stringify(MyBackgroundFetchProps)}</Text>;
};

describe('MyBackgroundFetch', () => {
  it('Use provider', async () => {
    const { getByTestId, toJSON } = render(
      <MyBackgroundFetchProvider backgroundFetchTaskName="taskName">
        <TestComponent />
      </MyBackgroundFetchProvider>,
    );
    await act(() => Promise.resolve());
    expect(getByTestId('props').props.children).toBe('{}');
    expect(toJSON()).toMatchSnapshot();
  });

  it('Use Hook', async () => {
    const wrapper = ({ children }: PropsWithChildren) => (
      <MyBackgroundFetchContext.Provider
        value={{ registered: true, status: BackgroundFetchStatus.Available }}
      >
        {children}
      </MyBackgroundFetchContext.Provider>
    );
    const { result } = renderHook(() => useMyBackgroundFetch(), { wrapper });
    await act(() => Promise.resolve());
    expect(result.current).toEqual({
      registered: true,
      status: BackgroundFetchStatus.Available,
    });
  });

  it('Use HoC', async () => {
    const TestedComponent = WithMyBackgroundFetch(TestComponent);
    const { getByTestId, toJSON } = render(
      <TestedComponent backgroundFetchTaskName="taskName" />,
    );
    await act(() => Promise.resolve());
    expect(getByTestId('props').props.children).toBe('{}');
    expect(toJSON()).toMatchSnapshot();
  });
});
