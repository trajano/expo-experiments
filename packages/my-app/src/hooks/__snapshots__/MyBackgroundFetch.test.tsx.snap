// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyBackgroundFetch Use HoC 1`] = `
<Text
  testID="props"
>
  {}
</Text>
`;

exports[`MyBackgroundFetch Use provider 1`] = `
<Text
  testID="props"
>
  {}
</Text>
`;
